src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/PostgresLedgerExtenderService.cs

```diff
@@ -1618,11 +1618,16 @@ void AggregateEntityResourceInternal(long entityId, long resourceEntityId)
 
                 void AggregateEntityResourceVaultInternal(long entityId, long resourceEntityId, long resourceVaultEntityId)
                 {
+                    // we only want to create new aggregate resource value history entry if
+                    // - given vault is seen for the very first time,
+                    // - given vault is already stored but has been updated and this update caused change of order (this is evaluated right before db persistence)
+
                     var lookup = new EntityResourceVaultLookup(entityId, resourceEntityId);
 
                     if (mostRecentEntityResourceVaultAggregateHistory.TryGetValue(lookup, out var existingAggregate))
                     {
-                        if (existingAggregate.VaultEntityIds.Contains(resourceVaultEntityId))
+                        // we're already the most recent one, there's nothing more to do
```

Comment:
Hmm this relates to my comments on the other file and the naming of last updated state version - which would make this early return incorrect.

Maybe a sensible performance improvement though 🤷 . I could be argued on this if we're willing to add a comment that it's not actually accurately the state versions.